src/main/java/week3/WordSearchTest.java

```diff
@@ -0,0 +1,131 @@
+package week3;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.util.HashSet;
+import org.junit.Before;
+import org.junit.Test;
+
+public class WordSearchTest {
+
+  private WordSearch wordSearch;
+
+  @Before
+  public void setUp() {
+    wordSearch = new WordSearch();
+  }
+
+  @Test
+  public void testEmptyDictionary() {
+    Dictionary d = new Dictionary(new String[]{});
+    char[][] grid = new char[][]{
+        {'A', 'A', 'R'},
+        {'T', 'C', 'D'}
+    };
+    assertEquals(new HashSet<String>(), wordSearch.findWords(grid, d));
+    assertEquals(new HashSet<String>(), wordSearch.findWords(null, d));
+    assertEquals(new HashSet<String>(), wordSearch.findWords(grid, null));
+  }
+
+  @Test
+  public void testEmptyGrid() {
+    Dictionary d = new Dictionary(new String[]{"CAR", "CARD", "CART", "CAT"});
+    char[][] grid = new char[0][0];
+
+    assertEquals(new HashSet<String>(), wordSearch.findWords(grid, d));
+  }
+
+  @Test
+  public void testExample() {
+    Dictionary d = new Dictionary(new String[]{"CAR", "CARD", "CART", "CAT"});
+    char[][] grid = new char[][]{
+        {'A', 'A', 'R'},
+        {'T', 'C', 'D'}
+    };
```

Comment:
You could pull out the re-used grids and dictionaries into shared variables, rather than C+Ping them.